WISE-Community/WISE
# Implemented authoring automated assessment structure

(needs UI improvements from @breity)

**Setup**
1. Add structures column to portal in the db.
```
ALTER TABLE `wise_database`.`portal` ADD COLUMN `structures` TEXT NULL AFTER `OPTLOCK`;
```
2. Import project that contains some automated assessments. Note the new project ID, which you'll use in the next step
[320.zip](https://github.com/WISE-Community/WISE/files/4768249/320.zip)

3. Set the ```automatedAssessmentProjectId``` field with the new project ID (replace NEWPROJECTID below)
```
UPDATE `wise_database`.`portal` SET `structures` = '{\"automatedAssessmentProjectId\":NEWPROJECTID}' WHERE (`id` = '1');
```

**Test**
1. Launch the AT, and go to structured authoring page. The new "Automated Assessment" option should appear. Click on it.
2. You should now see a list of steps from the newly-imported unit. For each step, you should be able to preview it, which launches in a new window.
3. Choose a step and go to the next page. You should now be able to customize the feedback.
4. After customizing, click next, and choose where to put the new step.

Test that the item is added to the unit, and scoring and feedback work as expected. If you want to add more assessment items, you can edit the imported unit. 

Resolves #2404
